mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java

```diff
@@ -38,7 +38,14 @@
  *
  */
 public interface MailboxMapper extends Mapper {
-    
+
+    /**
+     * Create the given {@link Mailbox} to the underlying storage
+     */
+    default MailboxId create(Mailbox mailbox) throws MailboxException {
+        return save(mailbox);
```

Comment:
Please add a precondition for verifying the mailboxId of the supplied mailbox is always null (and a test for it)

This will prevent using this method for renames (what you can do as off today)